include::../description.adoc[]

include::../recommended.adoc[]

== Noncompliant Code Example

[source,terraform]
----
resource "azurerm_role_definition" "example" { # Noncompliant
  name        = "example"
  scope       = data.azurerm_subscription.primary.id

  permissions {
    actions     = ["*"]
    not_actions = []
  }

  assignable_scopes = [ 
    data.azurerm_subscription.primary.id 
  ]
}
----

== Compliant Solution

[source,terraform]
----
resource "azurerm_role_definition" "example" {
  name        = "example
  scope       = data.azurerm_subscription.primary.id

  permissions {
    actions     = ["Microsoft.Compute/*"]
    not_actions = []
  }

  assignable_scopes = [ 
    data.azurerm_subscription.primary.id 
  ]
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]