include::../description.adoc[]


== Noncompliant Code Example

https://cryptography.io[cryptography]

[source,python]
----
def encrypt(key, ptxt):
    nonce   = '7cVgr5cbdCZV'.encode('utf-8') # The initialization vector is a static value

    cipher  = AESGCM(key)

    cipher.encrypt(nonce, ptxt, None) # Noncompliant
----

https://pycryptodome.readthedocs.io[pycryptodome]

[source,python]
----
def encrypt(key, ptxt):
    nonce   = '7cVgr5cbdCZV'.encode('utf-8') # The initialization vector is a static value

    cipher = AES.new(key, AES.MODE_CCM, nonce) # Initialization vector is configured here
    
    cipher.encrypt(ptxt) # Noncompliant
----

== Compliant Solution

https://cryptography.io[cryptography]

[source,python]
----
def encrypt(key, ptxt):
    nonce   = urandom(12) # Random 96 bit IV

    cipher  = AESGCM(key)s

    cipher.encrypt(nonce, ptxt, None)
----

https://pycryptodome.readthedocs.io[pycryptodome]

[source,python]
----
def encrypt(key, ptxt):
    nonce   = get_random_bytes(12) # Random 96 bit IV

    cipher  = AES.new(key, AES.MODE_CCM, nonce)

    cipher.encrypt_and_digest(ptxt)
----

include::../see.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::./message.adoc[]

include::./highlighting.adoc[]

endif::env-github,rspecator-view[]