include::../description.adoc[]

== Noncompliant Code Example

[source,python]
----
def do_something(a, b):  # `a` is unused
  return compute(b)
----

== Compliant Solution

[source,python]
----
def do_something(b):
  return compute(b)
----

== Exceptions

Functions in classes that override a class or implement interfaces are ignored.

[source,python]
----
class Parent(object):
  def get_value(self, a, b):
    return compute(a + b);

class Child(Patent):
  def get_value(self, a, b):
    return compute(b)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
