In regular expressions the boundary ``++^++`` can only match at the beginning of the input (or, in case it is combined with the multiline ``++m++`` flag, the beginning of the line) and ``++$++`` only at the end.


These patterns can be misused, by accidentally switching ``++^++`` and ``++$++`` for example, to create a pattern that can never match.


== Noncompliant Code Example

----
// This can never match because $ and ^ have been switched around
/$[a-z]+^/ // Noncompliant
----


== Compliant Solution

----
/^[a-z]+$/
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
