When using the imported value exclusively for typing annotations, use `import type` instead of `import`.

Type-only imports are always fully erased during transpilation, therefore it reduces the amount of code loaded at runtime.

Forcing type-only imports prevents issues where the file referenced in the import exports a type and the import is done for a value:

[source,javascript]
----
// ./types.ts
export type Car = 2

// ./main.ts
import { Car } from './types'
----

This would fail at transpilation, because the export would be erased and the import remain, thus breaking the chain.

== Noncompliant Code Example

[source,javascript]
----
import { Car } from 'vehicle';
const x: Car = 1;
----

== Compliant Solution

[source,javascript]
----
import type { Car } from 'vehicle';
const x: Car = 1;
----
