When using the imported value exclusively for typing annotations, use `import type` instead of `import`.

== Noncompliant Code Example

[source,javascript]
----
import { Foo } from 'FooBar';
const x: Foo = 1;
----

== Compliant Solution

[source,javascript]
----
import type { Foo } from 'FooBar';
const x: Foo = 1;
----
